modules/ppcp-api-client/src/ApiModule.php

```diff
@@ -133,6 +134,34 @@ static function () use ( $c ) {
 			}
 		);
 
+		/**
+		 * Filters the request arguments to add the `'PayPal-Partner-Attribution-Id'` header.
+		 *
+		 * This ensures that all API requests include the appropriate BN code retrieved
+		 * from the `PartnerAttribution` helper. Using this approach avoids the need
+		 * for extensive refactoring of existing classes that use the `RequestTrait`.
+		 *
+		 * The filter is applied in {@see RequestTrait::request()} before making an API request.
+		 *
+		 * @see PartnerAttribution::get_bn_code() Retrieves the BN code dynamically.
+		 * @see RequestTrait::request() Where the filter `ppcp_request_args` is applied.
+		 */
+		add_filter(
+			'ppcp_request_args',
+			static function ( array $args ) use ( $c ) {
+				if ( isset( $args['headers']['PayPal-Partner-Attribution-Id'] ) ) {
+					return $args;
+				}
+
+				$partner_attribution = $c->get( 'api.helper.partner-attribution' );
+				assert( $partner_attribution instanceof PartnerAttribution );
+
+				$args['headers']['PayPal-Partner-Attribution-Id'] = $partner_attribution->get_bn_code();
```

Comment:
Possibly not needed, but I would add a check to see if `'header'` actually exists before assigning the BN code:

```php
if (!isset($args['headers']) || !is_array($args['headers'])) {
    $args['headers'] = array();
}
```